File: ios/Classes/MainFlutterPlugin.swift
import Flutter
import UIKit

public class MainFlutterPlugin: NSObject, FlutterPlugin {
    private var plugins: [BaseFlutterPlugin] = []
    static var instance: MainFlutterPlugin?

    public static func register(with registrar: FlutterPluginRegistrar) {
        instance = MainFlutterPlugin()
        instance?.onAttachedToEngine(registrar: registrar)
    }

    private func onAttachedToEngine(registrar: FlutterPluginRegistrar) {
        plugins = [
            ZeticMLangeModelFlutterPlugin(
                registrar: registrar,
                channelName: "zetic_mlange_model_plugin"
            ),

            YOLOv8WrapperFlutterPlugin(
                registrar: registrar,
                channelName: "yolov8_plugin"
            )
        ]
    }

    deinit {
        for plugin in plugins {
            plugin.channel.setMethodCallHandler(nil)
        }
    }
}

File: android/src/main/kotlin/com/zeticai/zetic_mlange_flutter/wrapper/MainFlutterPlugin.kt
package com.zeticai.zetic_mlange_flutter.wrapper

import android.content.Context
import com.zeticai.zetic_mlange_flutter.core.BaseFlutterPlugin
import com.zeticai.zetic_mlange_flutter.core.FlutterPluginMethodMapper
import com.zeticai.zetic_mlange_flutter.wrapper.core.model.ZeticMLangeModelFlutterPlugin
import com.zeticai.zetic_mlange_flutter.wrapper.feature.yolov8.YOLOv8WrapperFlutterPlugin

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.plugin.common.MethodChannel.MethodCallHandler

class MainFlutterPlugin: FlutterPlugin {
    private var plugins: List<BaseFlutterPlugin<out Any>> = emptyList()

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        plugins = listOf(
            ZeticMLangeModelFlutterPlugin(flutterPluginBinding, "zetic_mlange_model_plugin"),
            YOLOv8WrapperFlutterPlugin(flutterPluginBinding, "yolov8_plugin"),
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        plugins.forEach {
            it.channel.setMethodCallHandler(null)
        }
    }
}